pkg/apis/extensions/v1beta1/conversion.go

```diff
@@ -27,6 +27,13 @@ import (
 	"k8s.io/kubernetes/pkg/conversion"
 	"k8s.io/kubernetes/pkg/runtime"
 	"k8s.io/kubernetes/pkg/util/intstr"
+	"k8s.io/kubernetes/pkg/util/labels"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+)
+
+const (
+	// annotation key used to identify non-convertible json paths.
+	NonConvertibleAnnotation = "kubernetes.io/non-convertible"
```

Comment:
> In the general case you need to handle having N failures.

I see. Traversing all the annotations and comparing the key with a prefix sounds ok.